homeassistant/components/openweathermap/__init__.py

```diff
@@ -1 +1,103 @@
 """The openweathermap component."""
+import asyncio
+import logging
+
+from pyowm import OWM
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import (
+    COMPONENTS,
+    CONF_LANGUAGE,
+    DOMAIN,
+    ENTRY_FORECAST_COORDINATOR,
+    ENTRY_NAME,
+    ENTRY_WEATHER_COORDINATOR,
+)
+from .forecast_update_coordinator import ForecastUpdateCoordinator
+from .weather_update_coordinator import WeatherUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict) -> bool:
+    """Set up the OpenWeatherMap component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
+    """Set up OpenWeatherMap as config entry."""
+    name = config_entry.data[CONF_NAME]
+    api_key = config_entry.data[CONF_API_KEY]
+    latitude = config_entry.data[CONF_LATITUDE]
+    longitude = config_entry.data[CONF_LONGITUDE]
+    forecast_mode = _get_config_value(config_entry, CONF_MODE)
+    language = _get_config_value(config_entry, CONF_LANGUAGE)
+
+    owm = OWM(API_key=api_key, language=language)
+    weather_coordinator = WeatherUpdateCoordinator(owm, latitude, longitude, hass)
+    forecast_coordinator = ForecastUpdateCoordinator(
+        owm, latitude, longitude, forecast_mode, hass
+    )
+
+    await weather_coordinator.async_refresh()
+    await forecast_coordinator.async_refresh()
+
+    if (
+        not weather_coordinator.last_update_success
+        and not forecast_coordinator.last_update_success
+    ):
+        raise ConfigEntryNotReady
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = {
+        ENTRY_NAME: name,
+        ENTRY_WEATHER_COORDINATOR: weather_coordinator,
+        ENTRY_FORECAST_COORDINATOR: forecast_coordinator,
+    }
+
+    for component in COMPONENTS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
+        )
```

Comment:
According to https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md data selectors should not be used unless there is a good reason like limiting API calls.  Is that the case here?

Reply:
@RobBie1221 yeah, I will add option to generate or not sensor entities
@MatthewFlamm I'm not sure what are you talking about, If you wish to send to user all raw information received in requests, I don't think this will be readable and useful.

Reply:
This ADR mentions monitored conditions as example, but this then would also apply when using multiple platforms? Generally speaking, a lot of integrations are out there using multiple platforms (like this sensor and weather). That essentially means when these type of integrations are moved to the UI, they need to setup all platforms unless separate API calls are required to get the data?
Thanks for clarifying!

Reply:
@freekode I think what @MatthewFlamm means is that disabling sensor platform with an option is under ADR-0003 not allowed unless separate API calls are needed to get the data.

Reply:
Well, for now looks like OWM makes different requests for forecast and weather.
But it will be rewritten for one-call API with new version of library.
So I think, we leave it as it is for now, because later this option anyway will be removed

Reply:
On the contrary, I'm saying that I think selecting data in the config flow isn't allowed and disabling entities like this is preferred, but the ADR isn't too clear on this regard.

Reply:
> That essentially means when these type of integrations are moved to the UI, they need to setup all platforms unless separate API calls are required to get the data?

That is my understanding, yes

Reply:
Ok clear enough, thanks!

Reply:
I am not the authority on this, just my reading of the ADR, which isn't too explicit on data selectors via platform vs in one platform.

Reply:
so, leave as it is